### PR TITLE
[IMPROVE] Check permission to create a room

### PR DESCRIPTION
**Pull request by reinaldonetof**
## Proposed changes

* Check if the user has permission to create each one inside the `NewMessageView`: Channel, Team, Direct Message, and Discussion
* In `RoomListView`check if the user has the permissions to create, and if has at least one permission show the button to create, else don't show the button
* In `CreateChannelView` check which permission the user has: if there is permission to create a public channel, a private channel, or both.

We're showing create team, channel, etc without checking if the user has the correct permission.
We should look for it first.

## How to test or reproduce

You need to be the Admin of server to test correctly this, so:

- 1st Create another account as User role and logs in from mobile

- then with the Admin's account go to `Administration > Permissions`, disable the permissions for user: `Create Public Channel`, `Create Private Channel`, `Create Team`, `Create Direct Messages`, `Start Discussion`
 -- In `RoomListView` the mobile's user shouldn't see the button to create a room when all those permissions are disabled for him. To show the button needs at least one of those permissions available.

- back to the Admin's account, you can disable/enable those permissions
 -- In NewMessageView for each permission you will see the types of rooms that you can create. To unauthorized the user to create a channel, the Admin's should disable `Create Public Channel`, `Create Private Channel` for him.

- Depend on the user's permission `Create Public Channel`, `Create Private Channel` will appear differently to create a channel.
* In `CreateChannelView` you will able to see if you can create a private, public channel, or both.

## Screenshots
<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</thead>
</tr>
<body>
<tr>
<td>

</td>
<td>

</td>
</tr>
<tr>
<td>

</td>
<td>

</td>
</tr>
<tr>
<td>

</td>
<td>

</td>
</tr>
</tbody>
</table>

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by gerzonc — APPROVED** [review]
LGTM

File: app/views/CreateChannelView.js [hunk 1]
```diff
@@ -102,14 +102,16 @@ class CreateChannelView extends React.Component {
 			readOnly: false,
 			encrypted: false,
 			broadcast: false,
-			isTeam
+			isTeam,
+			permissions: []
 		};
 		this.setHeader();
+		this.handleHasPermission();
```
Comment (gerzonc) [inline app/views/CreateChannelView.js:109]: Can you move it to a `componentDidMount` instead?

File: app/views/RoomsListView/index.js [hunk 2]
```diff
@@ -388,7 +398,7 @@ class RoomsListView extends React.Component {
 			},
 			headerRight: () => (searching ? null : (
 				<HeaderButton.Container>
-					{count > 0 ? (
+					{isDisabledToCreateRoom ? (
```
Comment (gerzonc) [inline app/views/RoomsListView/index.js:401]: Can you change this name to `canCreateRoom`? Feels more appealing

**Review by gerzonc — CHANGES_REQUESTED** [review]

**Review by gerzonc — APPROVED** [review]
LGTM!

**Review by gerzonc — APPROVED** [review]
LGTM

**Review by AlexAlexandre — APPROVED** [review]
LGTM :)

**Pull request merged**